src/views/TableView.vue

```diff
@@ -0,0 +1,145 @@
+<template>
+  <div class="container">
+    <div class="score-bar">Score Bar</div>
+    <div class="player1" v-if="updateTablePayload.players[0] !== ''">
+      {{ updateTablePayload.players[0] }}
+    </div>
+    <div class="player2" v-if="updateTablePayload.players[1] !== ''">
+      {{ updateTablePayload.players[1] }}
+    </div>
+    <div class="player3" v-if="updateTablePayload.players[2] !== ''">
+      {{ updateTablePayload.players[2] }}
+    </div>
+    <div class="player4" v-if="updateTablePayload.players[3] !== ''">
+      {{ updateTablePayload.players[3] }}
+    </div>
+    <div class="hand">
+      <div>
+        <div
+          class="card card1"
+          :class="{ transparent: updateTablePayload.cards[0].value === 0 }"
+        >
+          {{ updateTablePayload.cards[0].value
+          }}{{ updateTablePayload.cards[0].type }}
+        </div>
+        <div
+          class="card card4"
+          :class="{ transparent: updateTablePayload.cards[3].value === 0 }"
+        >
+          {{ updateTablePayload.cards[3].value
+          }}{{ updateTablePayload.cards[3].type }}
+        </div>
+        <div
+          class="card card3"
+          :class="{ transparent: updateTablePayload.cards[2].value === 0 }"
+        >
+          {{ updateTablePayload.cards[2].value
+          }}{{ updateTablePayload.cards[2].type }}
+        </div>
+        <div
+          class="card card2"
+          :class="{ transparent: updateTablePayload.cards[1].value === 0 }"
+        >
+          {{ updateTablePayload.cards[1].value
+          }}{{ updateTablePayload.cards[1].type }}
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import Component from "vue-class-component";
+
+@Component({
+  props: {
+    updateTablePayload1: {
+      type: Object
+    }
+  }
+})
+export default class TableView extends Vue {
+  // TODO remove test data and update props
+  updateTablePayload = {
+    cards: [
+      {
+        value: 12,
+        type: "Diamonds"
+      },
+      {
+        value: 13,
+        type: "Hearts"
+      },
+      {
+        value: 1,
+        type: "Clubs"
+      },
+      {
+        value: 12,
+        type: "Spades"
+      }
+    ],
+    gameScore: [1, 2],
+    sessionScore: [1, 2],
+    players: ["P1", "P2", "P3", "P4"]
+  };
```

Comment:
This is the data structure needed for the table. Everything has to have defaults except for `cards[x].type`

Reply:
How do you know who played each card if you removed playedBy property?